python/lsst/meas/base/simple_forced_measurement.py

```diff
@@ -0,0 +1,252 @@
+            measTable=self._finishOutputTable(table, measCat, exposure),
+        )
+
+    def _finishOutputTable(self, table, measCat, exposure):
+        """Finish the output table by adding the measurement results."""
+        measTable = measCat.asAstropy()
+        del measTable["id"]
+        del measTable["coord_ra"]
+        del measTable["coord_dec"]
+
+        measTable = astropy.table.hstack([table, measTable], join_type="exact")
+
+        # Add the X and Y pixel coordinates to the table.
+        measTable["x"], measTable["y"] = exposure.wcs.skyToPixelArray(
+            measTable[self.config.refCatRaColumn],
+            measTable[self.config.refCatDecColumn],
+            degrees=True,
+        )
+
+        return measTable
+
+    def _makeMinimalSourceCatalogFromAstropy(self, table):
+        """Create minimal schema SourceCatalog from an Astropy Table.
+
+        The forced measurement subtask expects this as input.
+
+        Parameters
+        ----------
+        table : `astropy.table.Table`
+            Table with locations and ids.
+
+        Returns
+        -------
+        outputCatalog : `lsst.afw.table.SourceTable`
+            Output catalog with minimal schema.
+        """
+        schema = lsst.afw.table.SourceTable.makeMinimalSchema()
+        outputCatalog = lsst.afw.table.SourceCatalog(schema)
+        outputCatalog.reserve(len(table))
+        # We should make the columns we grab here configurable.
+        for row in table:
+            objectId = row[self.config.refCatIdColumn]
+            ra = row[self.config.refCatRaColumn]
+            dec = row[self.config.refCatDecColumn]
+            outputRecord = outputCatalog.addNew()
+            outputRecord.setId(objectId)
+            outputRecord.setCoord(lsst.geom.SpherePoint(ra, dec, lsst.geom.degrees))
+        return outputCatalog
+
+    def _attachPsfShapeFootprints(self, sources, exposure, scaling=3):
+        """Attach Footprints to blank sources prior to measurement, by
+        creating elliptical Footprints from the PSF moments.
+
+        Parameters
+        ----------
+        sources : `lsst.afw.table.SourceCatalog`
+            Blank catalog (with all rows and columns, but values other than
+            ``coord_ra``, ``coord_dec`` unpopulated).
+            to which footprints should be attached.
+        exposure : `lsst.afw.image.Exposure`
+            Image object from which peak values and the PSF are obtained.
+        scaling : `int`, optional
+            Scaling factor to apply to the PSF second-moments ellipse in order
+            to determine the footprint boundary.
+        """
+        psf = exposure.getPsf()
+        if psf is None:
+            raise RuntimeError("Cannot construct Footprints from PSF shape without a PSF.")
+        bbox = exposure.getBBox()
+        wcs = exposure.getWcs()
+        for record in sources:
+            # TODO: vectorize this WCS call
+            localPoint = wcs.skyToPixel(record.getCoord())
```

Comment:
`wcs.skyToPixelArray(sources["ra"], sources["dec"], degrees=False)` (assuming radians).

You can also vectorize the contains call, I did that on the previous ticket.  `inBBox = np.atleast_1d(lsst.geom.Box2D(bbox).contains(x, y))` or something like that.

You still have to loop to make the footprints, I assume.